opts.pluginDefs is always nil, panics when using --plugin

I'm trying to use `--plugin`, but doing so breaks, as `opts.pluginDefs` is always nil, and never an allocated map.

```
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/jhump/goprotoc/app/goprotoc.parseFlags(0x0, 0x0, 0x7ffcc6e22009, 0x8, 0xc0000101f0, 0x5, 0x5, 0x7f7380, 0xc00000e018, 0xc000041e08, ...)
	/home/jake/go/pkg/mod/github.com/jhump/goprotoc@v0.2.0/app/goprotoc/options.go:174 +0x13b5
github.com/jhump/goprotoc/app/goprotoc.run(0xc0000101e0, 0x6, 0x6, 0x7f7360, 0xc00000e010, 0x7f7380, 0xc00000e018, 0x7f7380, 0xc00000e020, 0xc000073a01, ...)
	/home/jake/go/pkg/mod/github.com/jhump/goprotoc@v0.2.0/app/goprotoc/goprotoc.go:48 +0x157
github.com/jhump/goprotoc/app/goprotoc.Run(0xc0000101e0, 0x6, 0x6, 0x7f7360, 0xc00000e010, 0x7f7380, 0xc00000e018, 0x7f7380, 0xc00000e020, 0x6d1df7)
	/home/jake/go/pkg/mod/github.com/jhump/goprotoc@v0.2.0/app/goprotoc/goprotoc.go:35 +0x97
github.com/jhump/goprotoc/app/goprotoc.Main()
	/home/jake/go/pkg/mod/github.com/jhump/goprotoc@v0.2.0/app/goprotoc/goprotoc.go:30 +0x8a
main.main()
	/home/jake/go/pkg/mod/github.com/jhump/goprotoc@v0.2.0/cmd/goprotoc/goprotoc.go:22 +0x20
```

--- a/app/goprotoc/options.go
+++ b/app/goprotoc/options.go
@@ -171,6 +171,9 @@ func parseFlags(source string, programName string, args []string, stdout io.Writ
 				return fmt.Errorf("plugin name %s is not valid: name should have 'protoc-gen-' prefix", pluginName)
 			}
 			pluginName = pluginName[len("protoc-gen-"):]
+			if opts.pluginDefs == nil {
+				opts.pluginDefs = make(map[string]string, 1)
+			}
 			opts.pluginDefs[pluginName] = pluginLocation
 		default:
 			switch {
@@ -191,12 +194,12 @@ func parseFlags(source string, programName string, args []string, stdout io.Writ
 					return err
 				}
 			case strings.HasPrefix(parts[0], "--") && strings.HasSuffix(parts[0], "_out"):
-				if opts.output == nil {
-					opts.output = make(map[string]string, 1)
-				}
 				value, err := getOptionArg()
 				if err != nil {
 					return err
+				}
+				if opts.output == nil {
+					opts.output = make(map[string]string, 1)
 				}
 				opts.output[parts[0][2:len(parts[0])-4]] = value
 			default:

